### PR TITLE
Allow for manual donations

### PR DESCRIPTION
**Pull request by xdamman**
This PR enables orders to specify a "manual" payment method, in which case the order's status will be PENDING and will have to be manually confirmed as PAID.

Related issue: https://github.com/opencollective/opencollective/issues/1281

TODO:
- [x] New manual payment provider
- [x] Set default order status to PENDING
- [x] Email notification for pending orders

For another PR to avoid growing this PR even more:
- Add a mutation to mark a donation as PAID

### CODE REVIEW

**Comment by marcogbarcellos** [comment]
any status here?

File: test/utils.js [hunk 1]
```diff
@@ -172,6 +173,19 @@ export const readFee = (amount, feeStr) => {
   }
 };
 
+export const separator = length => {
+  const terminalCols =
+    process.platform === 'win32'
+      ? length || 40
+      : length || parseInt(execSync(`tput cols`).toString());
```
Comment (Betree) [inline test/utils.js]: Is this `separator` function used somewhere?

A user recently had a problem with `tput` not installed on an alpine linux (see https://github.com/opencollective/opencollective/issues/1443). Maybe we should encapsulate it in a separate `getTerminalCols` which would returns the default `40` if the call fails.

Comment (Betree) [reply test/utils.js]: Could happen on Windows machine (but this is tested here) or on Alpine linux if we ever run our tests in a such container (for example in CI). We could just put that in a `try..catch` block and return `40` if we catch.

Comment (xdamman) [reply test/utils.js]: Good idea. Done in this commit: https://github.com/opencollective/opencollective-api/pull/1448/commits/329526f32e1fee46de7081c103572ea06052065f

File: server/models/PaymentMethod.js [hunk 2]
```diff
@@ -308,9 +331,13 @@ export default function(Sequelize, DataTypes) {
...
       // If there is no monthly limit, the user needs to be an admin of the collective that owns the payment method
-      if (!this.monthlyLimitPerMember && !user.isAdmin(this.CollectiveId)) {
+      if (
+        !this.monthlyLimitPerMember &&
+        !user.isAdmin(this.CollectiveId) &&
+        this.type !== 'manual'
+      ) {
         throw new Error(
-          "You don't have enough permissions to use this payment method (you need to be an admin of the collective that owns this payment method)",
+          `You don't have enough permissions to use this payment method (you need to be an admin of the collective that owns this payment method)`,
```
Comment (marcogbarcellos) [inline server/models/PaymentMethod.js]: we better use single quotes when there's no variable to be inserted in the string template.

File: server/paymentProviders/opencollective/manual.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    netAmountInCollectiveCurrency: order.totalAmount - hostFeeInHostCurrency - platformFeeInHostCurrency,
+    amountInHostCurrency: order.totalAmount,
+    hostFeeInHostCurrency,
+    platformFeeInHostCurrency,
+    paymentProcessorFeeInHostCurrency,
+    description: order.description,
+  };
+
+  const creditTransaction = await models.Transaction.createFromPayload(payload);
+  console.log(">> creditTransaction", creditTransaction.dataValues);
```
Comment (marcogbarcellos) [inline server/paymentProviders/opencollective/manual.js]: no need for console log here, right?

File: test/utils.js [hunk 4]
```diff
@@ -172,6 +173,28 @@ export const readFee = (amount, feeStr) => {
...
+};
+
+export const separator = length => {
+  const terminalCols = length || getTerminalCols();
+
+  let separator = '';
+  for (let i = 0; i < terminalCols; i++) {
+    separator += '-';
+  }
+  console.log(`\n${separator}\n`);
```
Comment (marcogbarcellos) [inline test/utils.js]: perhaps remove this console log as well...

**Comment by marcogbarcellos** [comment]
other than the minor things on my comments, looks really good to me!

File: server/paymentProviders/opencollective/manual.js (hunk 3 above)
Comment (xdamman) [reply server/paymentProviders/opencollective/manual.js]: right (that's 2 month old but I'll take the opportunity to remove this)

File: test/utils.js (hunk 4 above)
Comment (xdamman) [reply test/utils.js]: well no, that's the whole point of this helper function: make it easier to debug in the console by spacing out and separating sections.

File: server/models/PaymentMethod.js (hunk 2 above)
Comment (xdamman) [reply server/models/PaymentMethod.js]: good catch. I'll use double quotes here because of the `don't`.

File: test/utils.js (hunk 4 above)
Comment (marcogbarcellos) [reply test/utils.js]: I see, makes sense! missed that

**Comment by znarf** [comment]
@xdamman for the "algorithm", I'm afraid the only thing that make sense is discussions between nice humans :-)

**Comment by xdamman** [comment]
@znarf ok, so based on this conversation, I'm merging this :-)

**Pull request merged**